bee-network/bee-autopeering/src/discovery/query.rs

```diff
@@ -0,0 +1,267 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    discovery::manager,
+    event::EventTx,
+    peer::{
+        lists::{ActivePeer, ActivePeersList, MasterPeersList, ReplacementList},
+        PeerId,
+    },
+    request::RequestManager,
+    server::ServerTx,
+    task::Repeat,
+};
+
+use rand::{thread_rng, Rng};
+
+#[derive(Clone)]
+pub(crate) struct QueryContext {
+    pub(crate) request_mngr: RequestManager,
+    pub(crate) master_peers: MasterPeersList,
+    pub(crate) active_peers: ActivePeersList,
+    pub(crate) replacements: ReplacementList,
+    pub(crate) server_tx: ServerTx,
+    pub(crate) event_tx: EventTx,
+}
+
+// Hive.go: pings the oldest active peer.
+pub(crate) fn do_reverify() -> Repeat<QueryContext> {
+    Box::new(|ctx| {
+        // Determine the next peer to re/verifiy.
+        if let Some(peer_id) = peer_to_reverify(&ctx.active_peers) {
+            log::debug!("Reverifying {}...", peer_id);
+
+            let ctx_ = ctx.clone();
+
+            // TODO: introduce `UnsupervisedTask` type, that always finishes after a timeout.
```

Comment:
In case you missed it

Reply:
I'ld keep the TODO for the 0.1.0 release as this is a more involved feature addition in the future that also depends on team decisions.